src/TestFramework/Factory.php

```diff
@@ -146,20 +146,17 @@ public function create(string $adapterName, bool $skipCoverage): TestFrameworkAd
     /**
      * Get only those source files that will be mutated to use them in coverage whitelist
      *
-     * @return list<string>
+     * @return list<SplFileInfo>
      */
     private function getFilteredSourceFilesToMutate(): array
     {
         if ($this->sourceFileFilter->getFilters() === []) {
             return [];
         }
 
-        /** @var list<string> $filteredPaths */
-        $filteredPaths = array_filter(array_map(
-            static fn (SplFileInfo $file) => $file->getRealPath(),
-            iterator_to_array($this->sourceFileFilter->filter($this->infectionConfig->getSourceFiles()))
-        ));
+        /** @var list<SplFileInfo> $files */
+        $files = iterator_to_array($this->sourceFileFilter->filter($this->infectionConfig->getSourceFiles()));
```

Comment:
could this change (that is a bit noisy as it is affecting a few method signatures / phpstan / mutations) be extracted?